python/hail/methods/family_methods.py

```diff
@@ -225,4 +216,207 @@ def mendel_errors(dataset, pedigree):
+    | HomVar | Het    | Het    | Auto       | 2 | 0 |
+    +--------+--------+--------+------------+---+---+
+    | HomVar | Het    | HomVar | Auto       | 1 | 0 |
+    +--------+--------+--------+------------+---+---+
+    | HomVar | HomVar | Het    | Auto       | 1 | 0 |
+    +--------+--------+--------+------------+---+---+
+    | HomRef | HomRef | Het    | HemiX      | 0 | 1 |
+    +--------+--------+--------+------------+---+---+
+    | HomRef | HomVar | Het    | HemiX      | 0 | 1 |
+    +--------+--------+--------+------------+---+---+
+    | HomVar | HomRef | Het    | HemiX      | 1 | 0 |
+    +--------+--------+--------+------------+---+---+
+    | HomVar | HomVar | Het    | HemiX      | 1 | 0 |
+    +--------+--------+--------+------------+---+---+
+
+    :func:`tdt` only considers complete trios (two parents and a proband with
+    defined sex).
+
+    :func:`tdt` produces a table with the following columns:
+
+     - **v** (*Variant*) -- Variant tested.
+
+     - **t** (*Int32*) -- Number of transmitted alternate alleles.
+
+     - **u** (*Int32*) -- Number of untransmitted alternate alleles.
+
+     - **chi2** (*Float64*) -- TDT statistic.
+
+     - **pval** (*Float64*) -- p-value.
+
+    Parameters
+    ----------
+    dataset : :class:`.MatrixTable`
+        Dataset.
+    pedigree : :class:`~hail.genetics.Pedigree`
+        Sample pedigree.
+
+    Returns
+    -------
+    :class:`.Table`
+        Table of TDT results.
+    """
+
+    hom_ref = 0
+    het = 1
+    hom_var = 2
+    
+    auto = 2
+    hemi_x = 1
+    
+    l = [(hom_ref,     het,     het,   auto, 0, 2),
+         (hom_ref, hom_ref,     het,   auto, 0, 1),
+         (hom_ref,     het, hom_ref,   auto, 0, 1),
+         (    het,     het,     het,   auto, 1, 1),
+         (    het, hom_ref,     het,   auto, 1, 0),
+         (    het,     het, hom_ref,   auto, 1, 0),
+         (    het, hom_var,     het,   auto, 0, 1),
+         (    het,     het, hom_var,   auto, 0, 1),
+         (hom_var,     het,     het,   auto, 2, 0),
+         (hom_var,     het, hom_var,   auto, 1, 0),
+         (hom_var, hom_var,     het,   auto, 1, 0),
+         (hom_ref, hom_ref,     het, hemi_x, 0, 1),
+         (hom_ref, hom_var,     het, hemi_x, 0, 1),
+         (hom_var, hom_ref,     het, hemi_x, 1, 0),
+         (hom_var, hom_var,     het, hemi_x, 1, 0)]
+    
+    mapping = {functions.capture([v[0], v[1], v[2], v[3]]): [v[4], v[5]] for v in l}
+    
+    tri = trio_matrix(dataset, pedigree, complete_trios=True)
+    tri = tri.filter_cols(functions.is_defined(tri.is_female))
+    tri = tri.annotate_globals(mapping = mapping)
+
+    tri = tri.annotate_rows(category =
+        functions.cond(tri.v.in_autosome_or_par(),
+                        0,
+                        functions.cond(tri.v.in_x_nonpar(),
+                                       1,
+                                       -1)))
+    
+    # FIXME calculate in one pass once aggs are re-vectorized
```

Comment:
? You should do this now, I don't see anywhere in the second annotate_rows that uses `tri.t`